UnityGLTF/Assets/UnityGLTF/Scripts/Extensions/GLTFSceneExporter.Animation.cs

```diff
@@ -0,0 +1,335 @@
+							break;
+
+						default:
+							throw new NotSupportedException();
+					}
+
+					//new channel: holds created sampler, and a new target
+					channels.Add(new AnimationChannel
+					{
+						Sampler = samplerId,
+						Target = new AnimationChannelTarget
+						{
+							Node = this._nodeCache[targetGameObject.transform],
+							Path = (GLTFAnimationChannelPath) property,
+						}
+					});
+				}
+			}
+
+			//exported channels and samplers for this animation clip, cache it and attach to GLTF root
+			index = _root.Animations.Count;
+			_animationCache.Add(unityAnimationClip, index);
+
+			var animation = new GLTF.Schema.Animation()
+			{
+				Name = unityAnimationClip.name,
+				Channels = channels,
+				Samplers = samplers
+			};
+			
+			_root.Animations.Add(animation);
+		}
+
+		//for all editor curve bindings of an animation clip, return a map: path -> (property -> member)
+		private Dictionary<string, Dictionary<string, Dictionary<string, EditorCurveBinding>>> GroupAnimationCurveBindings(IEnumerable<EditorCurveBinding> editorCurveBindings)
+		{
+			var bindings = new Dictionary<string, Dictionary<string, Dictionary<string, EditorCurveBinding>>>();
+
+			foreach (var editorCurveBinding in editorCurveBindings)
+			{
+				Dictionary<string, Dictionary<string, EditorCurveBinding>> propertyBindings;
+				if (!bindings.TryGetValue(editorCurveBinding.path, out propertyBindings))
+				{
+					propertyBindings = new Dictionary<string, Dictionary<string, EditorCurveBinding>>();
+					bindings.Add(editorCurveBinding.path, propertyBindings);
+				}
+
+				var split = editorCurveBinding.propertyName.Split(new[] { '.' }, 2);
+				var property = split[0];
+
+				Dictionary<string, EditorCurveBinding> memberBindings;
+				if (!propertyBindings.TryGetValue(property, out memberBindings))
+				{
+					memberBindings = new Dictionary<string, EditorCurveBinding>();
+					propertyBindings.Add(property, memberBindings);
+				}
+
+				var member = split[1];
+				memberBindings.Add(member, editorCurveBinding);
+			}
+
+			return bindings;
+		}
+
+		private AnimationSampler ExportAnimationSamplerPosition(AnimationCurve curveX, AnimationCurve curveY, AnimationCurve curveZ)
+		{
+			return this.ExportAnimationSampler(
+				new[] { curveX, curveY, curveZ },
+				keyIndex => GetRightHandedVector(new Vector3(curveX.keys[keyIndex].inTangent, curveY.keys[keyIndex].inTangent, curveZ.keys[keyIndex].inTangent)),
+				keyIndex => GetRightHandedVector(new Vector3(curveX.keys[keyIndex].value, curveY.keys[keyIndex].value, curveZ.keys[keyIndex].value)),
+				keyIndex => GetRightHandedVector(new Vector3(curveX.keys[keyIndex].outTangent, curveY.keys[keyIndex].outTangent, curveZ.keys[keyIndex].outTangent)),
+				time => GetRightHandedVector(new Vector3(curveX.Evaluate(time), curveY.Evaluate(time), curveZ.Evaluate(time))),
+				values => this.ExportData(values));
+		}
+
+		private static Quaternion CreateNormalizedQuaternion(float x, float y, float z, float w)
```

Comment:
this should already be implemented by unity

Reply:
from what I've seen, Unity expects the quaternions to be normalized and does not provide a method to check/do this.